### PR TITLE
Support QR decomposition on short-and-fat tensor(sfqr method)

### LINKED ISSUE #114: [TENSOR] QR decomposition for Short-and-Fat matrix
**Is your feature request related to a problem? Please describe.**
`mt.linalg.qr` now only works for tall and skinny matrix, short and fat matrix is easy to implement in theory, something should be metioned is that 'sfqr' method cannot be used for `mt.linalg.svd`.

**Describe the solution you'd like**
Short-and-Fat matrix A can be split like [A1, A2, A3, ...], we can choose A1 to be decomposed as a product A1 = Q1R1 of an orthogonal matrix Q1 and an upper triangular matrix R1, matrix Q1 as A's orthogonal matrix, then compute R ([R1, R2, R3, ...]) matrix by matrix Q1 to make sure A2=Q1R2, A3=Q1R3...

### PR DESCRIPTION
**Pull request by NJUPole**
## What do these changes do?

Add 'sfqr' method for `mt.linalg.qr`.

## Related issue number

resolve #114

### CODE REVIEW

**Comment by chineking** [comment]
Better to open a issue first.

File: mars/tensor/expressions/linalg/core.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 from ..core import TensorOperandMixin
 
 
-class TSQR(TensorOperandMixin):
+class QRBase(TensorOperandMixin):
```
Comment (wjsi) [inline mars/tensor/expressions/linalg/core.py]: It is better to create separate classes for different QR methods and dispatch in ``qr(a, method='method')``.

Comment (NJUPole) [reply mars/tensor/expressions/linalg/core.py]: `TensorQR` is a subclass of `QRBase` now, if create separate classes, each class may provide a staticmethod `tile` ?

Comment (chineking) [reply mars/tensor/expressions/linalg/core.py]: `tile` is a classmethod, this is not a issue

**Pull request merged**